feat: display latest dev version in settings for non-main builds

Sjoerd-Bo3 opened the pull request:
## Summary
- Displays the latest development version number in Settings for non-main builds
- Shows comparison between current and latest available dev version
- Helps developers and testers identify when updates are available

## Description
This enhancement improves version tracking for development builds by fetching and displaying the latest available dev version from GitHub. The feature:
- Only activates for non-main branches (dev, alpha builds)
- Shows clear version comparison in Settings
- Helps users know if they're running the latest development version
- Maintains clean UI with conditional display logic

## Implementation Details
1. **Version Checking**: Extended `AppVersionChecker` to fetch latest dev releases
2. **Conditional Display**: Only shows for non-main builds to avoid confusion
3. **UI Integration**: Clean presentation below current version in Settings
4. **Async Loading**: Non-blocking fetch that doesn't impact app startup

## Test plan
- [ ] Build and run from dev branch
- [ ] Navigate to Settings and verify current version displays
- [ ] Confirm latest dev version appears below (if different)
- [ ] Test with no network connection - should handle gracefully
- [ ] Build from main branch and verify dev version does NOT appear
- [ ] Test version comparison logic with various version number formats

## Screenshots
| Dev Build | Main Build |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/698f8def-b303-4ae7-af97-766cae7c84da) | ![image](https://github.com/user-attachments/assets/1315a9e6-b521-45ee-9bb8-986039d84f01) |
| Shows both current and latest dev | Only shows current version |

Sjoerd-Bo3 committed:
feat: Display latest dev version in Settings for non-main builds

Add display of latest dev version (x.x.x.x format) in Settings for all non-main branch builds.
Shows version with visual indicators for update status. Uses same
caching and comparison logic as main version checking.

- Only visible when not on main branch builds
- Orange color with arrow icon when newer version available
- Secondary color with hammer icon when on latest version
- 24-hour cache to minimize API calls
- Properly parses APP_DEV_VERSION for x.x.x.x format
- Added localization entries for new strings

marionbarker reviewed (approved):
Reviewed and tested. Left comment questioning comment format.

Sjoerd-Bo3 committed:
style: Convert documentation comments to // style in AppVersionChecker

- Replaced all /// single-line documentation comments with //
- Converted multi-line /** */ comments to // style
- Maintains consistency with Trio's predominant comment style
- Addresses PR #582 review feedback about comment formatting

Sjoerd-Bo3 commented:
> ## Review and Test
> I plan to approve this from review and test.
> 
> ## Formatting Comment
> I noticed when reviewing that the comments in this PR use
> 
> /** comment */
> 
> Most files in Trio use // on each line that is a comment. That format is my personal preference because when searching for content, it is immediately obvious if a line is code or comment.
> 
> I assume there is a preferred style for Trio.
> 
> * If this is an accepted format, then no worries.
> * If the version of comments is desired to be consistent, then the list of "different" files is provided at the end of this comment; and would be the subject of a separate PR.
> 
> This command
> 
> ```
> find Trio/Sources -name "*.swift" -exec grep '//' {} \; | wc  
> ```
> 
> indicates // is used for 5040 lines, whereas /** is used 30 times.
> 
> These are the Trio/Sources files that contain instances of `/**`.
> 
> ```
> Trio/Sources/APS/Storage/CarbsStorage.swift
> Trio/Sources/Shortcuts/Override/OverridePresetsIntentRequest.swift
> Trio/Sources/Modules/Home/View/Chart/ChartElements/GlucoseTargetsView.swift
> Trio/Sources/Modules/Home/HomeStateModel+Setup/GlucoseTargetSetup.swift
> Trio/Sources/Views/TagCloudView.swift
> Trio/Sources/Services/Network/Nightscout/NightscoutManager.swift
> Trio/Sources/Services/Storage/Keychain/KeychainItemAccessibility.swift
> Trio/Sources/Services/Storage/Disk/Disk.swift
> Trio/Sources/Services/AppVersionChecker/AppVersionChecker.swift
> ```

@marionbarker Fixed

marionbarker reviewed (approved):
Test again after commit `71a278b`.
Works as expected for branches named `main` and `pr582` (with and without a modification to the APP_DEV_VERSION).

kingst commented on Trio/Sources/Services/AppVersionChecker/AppVersionChecker.swift:
```diff
@@ -1,75 +1,85 @@
 
+    // Dev version properties
+    // Cached app version for which dev data was last fetched.
+    @Persisted(key: "cachedForDevVersion") private var cachedForDevVersion: String? = nil
```
Just curious -- why are you caching this? Is there an issue with re-checking it or perhaps we could cache it in memory instead of persisting it. I can see that the rest of this module is persisting its properties, so more curious about the overall design.

kingst replied:
I got an answer for this one from @dnzxy via DM, so resolving.

kingst commented on Trio/Sources/Services/AppVersionChecker/AppVersionChecker.swift:
```diff
@@ -135,20 +143,92 @@ final class AppVersionChecker {
+    // - Parameter completion: A closure that receives:
+    // - latestDevVersion: The latest dev version string (if available).
+    // - isNewer: `true` if the fetched dev version is newer than the current version.
+    func checkForNewDevVersion(completion: @escaping (String?, Bool) -> Void) {
```
What do you think about using an `async` version instead? Most of the new code I've seen uses this newer pattern.

Sjoerd-Bo3 replied:
Fixed @kingst

kingst commented on Trio/Sources/Services/AppVersionChecker/AppVersionChecker.swift:
```diff
@@ -135,20 +143,92 @@ final class AppVersionChecker {
+    // - completion: A closure that receives:
+    // - latestDevVersion: The latest dev version string from GitHub (if available).
+    // - isNewer: `true` if the fetched dev version is newer than the current version.
+    private func fetchDevVersionAndUpdateCache(currentVersion: String, completion: @escaping (String?, Bool) -> Void) {
```
same async comment here

kingst commented on Trio/Sources/Services/AppVersionChecker/AppVersionChecker.swift:
```diff
@@ -259,19 +335,17 @@ final class AppVersionChecker {
         let lines = contents.split(separator: "\n")
         for line in lines {
-            if line.contains("APP_VERSION") {
+            if line.contains("APP_VERSION"), !line.contains("APP_DEV_VERSION") {
```
Do you need this second clause? At least in my config it seems like it's not needed

kingst reviewed (commented):
I left a few questions / comments / suggestions. I'm not that familiar with this module but hopefully my feedback is useful.

dnzxy commented:
We'd like to move to `0.5.1` this week. What's the status here @Sjoerd-Bo3 ?

Sjoerd-Bo3 commented:
> We'd like to move to `0.5.1` this week. What's the status here @Sjoerd-Bo3 ? Please also fix merge conflicts and sync your feature branch with latest `dev`. Thanks :)

I have the changes locally. Will push when home.

Sjoerd-Bo3 committed:
Merge branch 'dev' into feature/dev-version-display

Signed-off-by: Sjoerd Bozon <sjoerd.bozon@gmail.com>

Sjoerd-Bo3 committed:
Convert dev version methods to async/await

- Convert refreshVersionInfo to async/await with completion handler wrapper
- Convert private checkForNewVersion to async/await with completion handler wrapper
- Convert fetchDataAndUpdateCache to async/await with parallel data fetching
- Convert checkAndNotifyVersionStatus to use async/await with @MainActor
- Update SettingsRootView to use async methods with Task blocks
- Simplify parseVersionFromConfig condition by checking for "DEV" instead of "APP_DEV_VERSION"

Sjoerd-Bo3 committed:
Complete async/await conversion for dev version methods

- Add async version of fetchData method
- Convert checkForNewDevVersion to async/await with completion handler wrapper
- Convert fetchDevVersionAndUpdateCache to async/await

Sjoerd-Bo3 commented:
Addressed Comments from @kingst, ready for review and merge

MikePlante1 reviewed (approved):
Approval based on testing, not code review:

| installed branch name | APP_VERSION | APP_DEV_VERSION | latest version (0.2.7)| latest dev (0.5.1.4) | new version alert |
| --- | --- | --- | --- | --- | --- |
| main | 0.2.6 | 0.2.6 | orange |   | yes |
| main | 0.2.7 | 0.2.7 | green |   | no |
| dev | 0.2.7 | 0.2.7 | green |  orange | no |
| test | 0.2.6 | 0.2.6 | orange | orange  | yes |
| test | 0.2.7 | 0.2.7 | green |  orange | no |
| test | 0.5.1 | 0.5.1.3 | green | orange  | no |
| test | 0.5.1 | 0.5.1.4 | green | gray  | no |
